From 3cf2da6df9643a0856ca13d8cb58ba37ab6ed174 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Wed, 5 May 2021 14:28:22 +0200
Subject: [PATCH 115/841] feat(st-sdmmc2): set FIFO size to 1024 on STM32MP25

On STM32MP25, a new version of the SDMMC2 IP is embedded (v3.0).
The size of the FIFO is 1024 in this new IP version.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 90f9cd66a85ad93288e76ff3a32833761fa1a3f3 feat(st-sdmmc2): set FIFO size to 1024 on STM32MP25]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: Ie6b1fb215fc77b24b7c342d4cd69248a96039a4d
---
 drivers/st/mmc/stm32_sdmmc2.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/st/mmc/stm32_sdmmc2.c b/drivers/st/mmc/stm32_sdmmc2.c
index be722f322..2cfa2e9bd 100644
--- a/drivers/st/mmc/stm32_sdmmc2.c
+++ b/drivers/st/mmc/stm32_sdmmc2.c
@@ -129,7 +129,12 @@
 #define DT_SDMMC2_COMPAT		"st,stm32-sdmmc2"
 #endif
 
+#if STM32MP13 || STM32MP15
 #define SDMMC_FIFO_SIZE			64U
+#endif
+#if STM32MP25
+#define SDMMC_FIFO_SIZE			1024U
+#endif
 
 #define STM32MP_MMC_INIT_FREQ			U(400000)	/*400 KHz*/
 #define STM32MP_SD_NORMAL_SPEED_MAX_FREQ	U(25000000)	/*25 MHz*/
-- 
2.47.2

